app/src/main/java/com/android/voyageur/ui/trip/schedule/WeeklyView.kt

```diff
@@ -0,0 +1,179 @@
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import com.android.voyageur.model.activity.Activity
+import com.android.voyageur.model.trip.Trip
+import com.android.voyageur.ui.navigation.BottomNavigationMenu
+import com.android.voyageur.ui.navigation.LIST_TOP_LEVEL_DESTINATION
+import com.android.voyageur.ui.navigation.NavigationActions
+import com.google.firebase.Timestamp
+import java.time.DayOfWeek
+import java.time.LocalDate
+import java.time.ZoneId
+
+@Composable
+fun WeeklyViewScreen(
+    trip: Trip,
+    navigationActions: NavigationActions,
+    onDaySelected: () -> Unit // Add this parameter
+) {
+  val weeks = generateWeeks(trip.startDate, trip.endDate)
+
+  Scaffold(
+      modifier = Modifier.fillMaxSize().testTag("weeklyViewScreen"),
+      bottomBar = {
+        BottomNavigationMenu(
+            onTabSelect = { route -> navigationActions.navigateTo(route) },
+            tabList = LIST_TOP_LEVEL_DESTINATION,
+            selectedItem = navigationActions.currentRoute())
+      }) { pd ->
+        Box(modifier = Modifier.fillMaxSize().padding(pd)) {
+          LazyColumn(
+              modifier = Modifier.fillMaxWidth().testTag("weeksColumn"),
+              verticalArrangement = Arrangement.spacedBy(16.dp),
+              horizontalAlignment = Alignment.CenterHorizontally) {
+                item { Spacer(modifier = Modifier.height(16.dp)) }
+                items(weeks.size) { weekIndex ->
+                  WeekCard(
+                      weekStart = weeks[weekIndex].first,
+                      weekEnd = weeks[weekIndex].last,
+                      activities = trip.activities,
+                      navigationActions = navigationActions,
+                      trip = trip,
+                      weekIndex = weekIndex,
+                      onDaySelected = onDaySelected // Pass the callback
+                      )
+                }
+              }
+        }
+      }
+}
+
+@Composable
+private fun WeekCard(
+    weekStart: LocalDate,
+    weekEnd: LocalDate,
+    activities: List<Activity>,
+    navigationActions: NavigationActions,
+    trip: Trip,
+    weekIndex: Int,
+    onDaySelected: () -> Unit // Add this parameter
+) {
+  Card(
+      modifier = Modifier.width(360.dp).testTag("weekCard_$weekIndex"),
```

Comment:
Might not adapt well to all screen sizes due to the hard coded size, you can use something like fillMaxWidth

Reply:
Thank you for the remark! I agree with you, I didn't think of it at first. I will fix it in the next PR.